### ISSUE TITLE
Lambda case quick action format instance heads

### ISSUE THREAD

**Issue by vladciobanu** [haskell/haskell-language-server#1674]
### Your environment
Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`:

```sh
haskell-language-server version: 1.0.0.0 (GHC: 8.10.2) (PATH: /nix/store/b3jr5z4hzv4vsysi0g7r1dx2a9ndi1ja-haskell-language-server-1.0.0.0/bin/haskell-language-server-wrapper)
Tool versions found on the $PATH
cabal:		3.4.0.0
stack:		2.5.1.1
ghc:		8.10.2
```
Which OS do you use: NixOS

Which lsp-client do you use: emacs

### Steps to reproduce

When filling in a typed hole in a class method with a lambda case split, the whole instance head/decl gets reformatted. This project has stylish-haskell setup and does not touch instances.

### Expected behaviour

Use the code formatter or not touch the instance head.

E.g., for this code:

```haskell
instance
  ( Cast (GBoolExp b a) (GBoolExp b' a')
  , Cast a a'
  ) => Cast (GBoolExp b a) (GBoolExp b' a') where
    cast = _
```

When doing "lambda case split" on the `_`, I get this:

```haskell
instance (Cast (GBoolExp b a) (GBoolExp b' a'), Cast a a') =>
         Cast (GBoolExp b a) (GBoolExp b' a') where
  cast
    = \case
        (BoolAnd l_gb'a') -> _
        (BoolOr l_gb'a') -> _
        (BoolNot gb'a') -> _
        (BoolExists gb'a') -> _
        (BoolFld a') -> _
```

What I would expect is this:

```haskell
instance
  ( Cast (GBoolExp b a) (GBoolExp b' a')
  , Cast a a'
  ) => Cast (GBoolExp b a) (GBoolExp b' a') where
  cast
    = \case
        (BoolAnd l_gb'a') -> _
        (BoolOr l_gb'a') -> _
        (BoolNot gb'a') -> _
        (BoolExists gb'a') -> _
        (BoolFld a') -> _
```

**Comment by googleson78** [comment]
This is also an issue when introducing a lambda or a lambda case split in functions with top leve pattern matching. If I introduce a lambda in the second case of such a function, hls will also reformat my first case

**Comment by isovector** [comment]
I am 95% sure this is fixed in HEAD (via https://github.com/haskell/haskell-language-server/pull/1543). Mind compiling from source?

**Comment by jneira** [comment]
The fix should be included in the hls-1.1.0 release, @vladciobanu @googleson78 feel free to reopen if it does not fix the issue for you
